web/src/buddy_list.js

```diff
@@ -79,51 +86,106 @@ export class BuddyList extends BuddyListConf {
+        // those headers then we show the total user count in the main title.
+        const default_userlist_title = $t({defaultMessage: "USERS"});
+        if (hide_headers) {
+            const total_user_count = people.get_active_human_count();
+            const formatted_count = message_view_header.get_formatted_sub_count(total_user_count);
+            const userlist_title = `${default_userlist_title} (${formatted_count})`;
+            $("#userlist-title").text(userlist_title);
             return;
         }
-        let subscriber_count;
+        $("#userlist-title").text(default_userlist_title);
+
+        let header_text;
+        let tooltip_text;
+
         if (current_sub) {
-            subscriber_count = peer_data.get_subscriber_count(current_sub.stream_id, false);
+            header_text = $t({defaultMessage: "In this stream"});
+            tooltip_text = $t(
+                {defaultMessage: "{N, plural, one {# subscriber} other {# subscribers}}"},
+                {N: subscriber_count},
+            );
         } else {
-            const pm_ids_list = pm_ids_string
-                ? people.user_ids_string_to_ids_array(pm_ids_string)
-                : [];
-            // Plus one for the "me" user, who isn't in the recipients list.
-            subscriber_count = pm_ids_list.length + 1;
+            header_text = $t({defaultMessage: "In this conversation"});
+            tooltip_text = $t(
+                {defaultMessage: "{N, plural, one {# participant} other {# participants}}"},
+                {N: subscriber_count},
+            );
         }
 
-        const has_inactive_users_matching_narrow =
-            subscriber_count > this.users_matching_view_ids.length;
-        const total_user_count = people.get_active_human_count();
-        const other_users_count = total_user_count - subscriber_count;
-        const has_inactive_other_users = other_users_count > this.other_user_ids.length;
+        $("#buddy-list-users-matching-view-container .buddy-list-subsection-header").append(
+            render_section_header({
+                id: "buddy-list-users-matching-view-section-heading",
+                header_text,
+                user_count: message_view_header.get_formatted_sub_count(subscriber_count),
+                tooltip_text,
+            }),
+        );
 
-        buddy_data.update_empty_list_placeholders(
-            has_inactive_users_matching_narrow,
-            has_inactive_other_users,
+        $("#buddy-list-other-users-container .buddy-list-subsection-header").append(
+            render_section_header({
+                id: "buddy-list-other-users-section-heading",
+                header_text: $t({defaultMessage: "Others"}),
+                user_count: message_view_header.get_formatted_sub_count(other_users_count),
+                tooltip_text: $t(
+                    {defaultMessage: "{N, plural, one {# other user} other {# other users}}"},
+                    {N: other_users_count},
```

Comment:
Maybe it would be better to move the `plural` thing into handlebars than here?

Reply:
I expect so.

Reply:
Oh actually maybe not, because we need to differentiate between "participant", "user", and "subscriber". I'll leave it as is for now